src/pgwire/src/protocol.rs

```diff
@@ -902,11 +925,11 @@ where
                     let rows = rows?;
                     count += rows.len();
                     for row in rows {
+                        encode_fn(row, typ, &mut out)?;
                         self.conn
-                            .send(BackendMessage::CopyData(message::encode_copy_row_text(
-                                row, typ,
-                            )))
+                            .send(BackendMessage::CopyData(out.clone()))
                             .await?;
+                        out.clear();
```

Comment:
Is this a perf problem? My suspicion is that it's at least as fast as the previous method (having the encode func return a vec) because (I believe) `Vec<u8>::clone` is a very fast mem copy, and after a row or two this vec will have sufficient capacity to not need to reallocate, so the additional copy time is perhaps less than any possible reallocs that would be occurring every row. Am I making that up?